src/comodo/jaxsimSimulator/jaxsimSimulator.py

```diff
@@ -0,0 +1,570 @@
+    logger.handlers.clear()
+logger.propagate = False
+# Console handler
+console_handler = logging.StreamHandler()
+formatter = logging.Formatter(
+    "[%(asctime)-19s.%(msecs)03d] [%(levelname)-8s] [TID %(thread)-5d] - %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+console_handler.setFormatter(formatter)
+logger.addHandler(console_handler)
+
+
+class JaxsimContactModelEnum(enum.IntEnum):
+    RIGID = enum.auto()
+    RELAXED_RIGID = enum.auto()
+    VISCO_ELASTIC = enum.auto()
+    SOFT = enum.auto()
+
+
+class JaxsimSimulator(Simulator):
+    def __init__(self) -> None:
+        # Flag to check if the simulator is initialized
+        self._is_initialized: bool = False
+
+        # Simulation data
+        self._data: js.data.JaxSimModelData | None = None
+
+        # Simulation model
+        self._model: js.model.JaxSimModel | None = None
+
+        # Integrator (not used for visco-elastic contacts)
+        self._integrator: integrators.common.Integrator | None = None
+
+        # Integrator state for the simulation (not used for visco-elastic contacts)
+        self._integrator_state: dict[str, Any] | None = None
+
+        # Time step for the simulation
+        self._dt: float | None = None
+
+        # Simulation time (seconds)
+        self._t: float = 0.0
+
+        # Joint torques
+        self._tau: npt.ArrayLike | None = None
+
+        # Link contact forces
+        self._link_contact_forces: npt.ArrayLike | None = None
+
+        # Contact model to use
+        self._contact_model_type: JaxsimContactModelEnum = (
+            JaxsimContactModelEnum.RELAXED_RIGID
+        )
+
+        # Index of the left foot link
+        self._left_foot_link_idx: int | None = None
+
+        # Index of the right foot link
+        self._right_foot_link_idx: int | None = None
+
+        # Index of the left foot sole frame
+        self._left_footsole_frame_idx: int | None = None
+
+        # Index of the right foot sole frame
+        self._right_footsole_frame_idx: int | None = None
+
+        # Mapping between user provided joint names and JaxSim joint names
+        self._to_js: npt.ArrayLike | None = None
+
+        # Mapping between JaxSim joint names and user provided joint names
+        self._to_user: npt.ArrayLike | None = None
+
+        # ==== Visualization attributes ====
+        # Mode for visualization ('record', 'interactive', or None)
+        self._visualization_mode: str | None = None
+
+        # Visualization object
+        self._viz: MujocoVisualizer | None = None
+
+        # Recorder for video recording
+        self._recorder: MujocoVideoRecorder | None = None
+
+        # Frames per second for visualization
+        self._viz_fps: int = 10
+
+        # Last rendered time (seconds)
+        self._last_rendered_t: float = 0.0
+
+        # Optional Mujoco model helper
+        self._mj_model_helper: MujocoModelHelper | None = None
+
+    # ==== Simulator interface methods ====
+
+    def load_model(
```

Comment:
If possible I would divide the operations needed for instantiate the simualtor (e.g. defintion of timestep and contact model), from loading the model itself (loading the robot model, the left foot link name etc). 
The reason for this is that Comodo is intended to be used for co-design, think about having a GA running where each chromosome represent a different robot model, then having clear distinction from what is related to the robot model (and needs to be updated at each loop) and what is not can minimize the errors.

Reply:
Thanks for the feedback! Do you mean to move some attributes of the simulator (like the dt and contact model type) to the constructor of the simulator class and leave to the `load_model` function just the arguments related to the model?

Reply:
Yes ! Either this or to have two distinct function, like  `initilize_simulator` and `load_model`

Reply:
Addressed in 9aa12e0c4d85abb3b37e9838ff4b076d9bd5447d

In the end I moved those parameters to the constructor to not add another function that is not part of the `Simulator` abstract class (we can think of update it in the future).